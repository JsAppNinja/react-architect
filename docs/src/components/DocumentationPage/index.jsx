import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import Helmet from 'react-helmet';
import CircularProgress from 'react-md/lib/Progress/CircularProgress';

const ACCESSIBILITY_PROPS = {
  'aria-describedby': 'loading-documentation',
  'aria-busy': true,
};

const DocumentationPage = ({ loading, children, title, className, ...otherProps }) => {
  let content = children;
  let props;
  if (loading) {
    content = <CircularProgress id="loading-documentation" key="loading" />;
    props = ACCESSIBILITY_PROPS;
  }

  return (
    <section {...props} {...otherProps} className={cn('md-grid md-grid--40-16', className)}>
      <Helmet title={title} />
      {content}
    </section>
  );
};

DocumentationPage.propTypes = {
  loading: PropTypes.bool.isRequired,
  children: PropTypes.node,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default DocumentationPage;
