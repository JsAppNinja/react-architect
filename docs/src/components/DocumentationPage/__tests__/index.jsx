/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import DocumentationPage from '../';

describe('DocumentationPage', () => {
  it('should render correctly based on the loading prop', () => {
    let tree = renderer.create(<DocumentationPage loading title="Hello" />).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <DocumentationPage
        title="Hello"
        loading={false}
      >
        <h1>Hello, World!</h1>
      </DocumentationPage>
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <DocumentationPage
        title="Hello"
        loading
      >
        <h1>Hello, World!</h1>
      </DocumentationPage>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
