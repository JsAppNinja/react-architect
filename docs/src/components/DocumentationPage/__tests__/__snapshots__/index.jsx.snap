// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentationPage should render correctly based on the loading prop 1`] = `
<section
  aria-busy={true}
  aria-describedby="loading-documentation"
  className="md-grid md-grid--40-16"
>
  <svg
    aria-valuemax={100}
    aria-valuemin={0}
    className="md-progress md-progress--circular md-block-centered md-progress--circular-indeterminate"
    height={24}
    id="loading-documentation"
    role="progressbar"
    style={Object {}}
    viewBox="0 0 66 66"
    width={24}
  >
    <circle
      className="md-circular-progress-path md-circular-progress-path--animated"
      cx="33"
      cy="33"
      r="30"
      strokeLinecap="round"
      strokeWidth="6"
      style={undefined}
    />
  </svg>
</section>
`;

exports[`DocumentationPage should render correctly based on the loading prop 2`] = `
<section
  className="md-grid md-grid--40-16"
>
  <h1>
    Hello, World!
  </h1>
</section>
`;

exports[`DocumentationPage should render correctly based on the loading prop 3`] = `
<section
  aria-busy={true}
  aria-describedby="loading-documentation"
  className="md-grid md-grid--40-16"
>
  <svg
    aria-valuemax={100}
    aria-valuemin={0}
    className="md-progress md-progress--circular md-block-centered md-progress--circular-indeterminate"
    height={24}
    id="loading-documentation"
    role="progressbar"
    style={Object {}}
    viewBox="0 0 66 66"
    width={24}
  >
    <circle
      className="md-circular-progress-path md-circular-progress-path--animated"
      cx="33"
      cy="33"
      r="30"
      strokeLinecap="round"
      strokeWidth="6"
      style={undefined}
    />
  </svg>
</section>
`;
